File: app/src/main/java/com/aquamorph/frcmanager/Adapters/EventSpinnerAdapter.java
package com.aquamorph.frcmanager.adapters;

import android.app.Activity;
import android.graphics.Color;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.SpinnerAdapter;
import android.widget.TextView;

import com.aquamorph.frcmanager.models.Events;

import java.util.ArrayList;

/**
 * <p></p>
 *
 * @author Christian Colglazier
 * @version 1/22/2016
 */
public class EventSpinnerAdapter extends BaseAdapter implements SpinnerAdapter {

	private final ArrayList<Events> eventList;
	private Activity activity;

	public EventSpinnerAdapter(ArrayList<Events> eventList, Activity activity) {
		this.eventList = eventList;
		this.activity = activity;
	}

	@Override
	public int getCount() {
		return eventList.size();
	}

	@Override
	public Object getItem(int position) {
		return eventList.get(position).name;
	}

	@Override
	public long getItemId(int position) {
		return position;
	}

	@Override
	public View getView(int position, View convertView, ViewGroup parent) {
		TextView text;
		if (convertView != null){
			// Re-use the recycled view here!
			text = (TextView) convertView;
		} else {
			// No recycled view, inflate the "original" from the platform:
			text = (TextView) activity.getLayoutInflater().inflate(
					android.R.layout.simple_dropdown_item_1line, parent, false
			);
		}
		text.setTextColor(Color.BLACK);
		text.setText(eventList.get(position).name);
		return text;
	}
}

File: app/src/main/kotlin/com/aquamorph/frcmanager/adapters/EventSpinnerAdapter.kt
package com.aquamorph.frcmanager.adapters

import android.app.Activity
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.SpinnerAdapter
import android.widget.TextView
import com.aquamorph.frcmanager.models.Event
import java.util.*

/**
 * Populates the spinner for the setup screen with a list of events a team is registered for.
 *
 * @author Christian Colglazier
 * @version 3/31/2018
 */
class EventSpinnerAdapter(private val eventList: ArrayList<Event>, private val activity: Activity) : BaseAdapter(), SpinnerAdapter {

    override fun getCount(): Int {
        return eventList.size
    }

    override fun getItem(position: Int): Any {
        return eventList[position].short_name
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        val text: TextView
        if (convertView != null) {
            // Re-use the recycled view here!
            text = (convertView as TextView?)!!
        } else {
            // No recycled view, inflate the "original" from the platform:
            text = activity.layoutInflater.inflate(
                    android.R.layout.simple_dropdown_item_1line, parent, false
            ) as TextView
        }
        text.text = eventList[position].name
        return text
    }


}

